code/modules/mob/living/living.dm

```diff
@@ -546,67 +558,29 @@ Sorry Giacom. Please don't be mad :(
 		else
 			return 0
 
-	if (restrained())
+	var/atom/movable/pullee = pulling
+	if(pullee && get_dist(src, pullee) > 1)
 		stop_pulling()
-
-
-	var/cuff_dragged = 0
-	if (restrained())
-		for(var/mob/living/M in range(1, src))
-			if (M.pulling == src && !M.incapacitated())
-				cuff_dragged = 1
-	if (!cuff_dragged && pulling && !throwing && (get_dist(src, pulling) <= 1 || pulling.loc == loc))
-		var/turf/T = loc
-		. = ..()
-
-		if (pulling && pulling.loc)
-			if(!isturf(pulling.loc))
-				stop_pulling()
-				return
-			else
-				if(Debug)
-					diary <<"pulling disappeared? at [__LINE__] in mob.dm - pulling = [pulling]"
-					diary <<"REPORT THIS"
-
-		/////
-		if(pulling && pulling.anchored)
+	var/turf/T = loc
+	. = ..()
+	if(. && pulling && pulling == pullee) //we were pulling a thing and didn't lose it during our move.
+		if(pulling.anchored)
 			stop_pulling()
 			return
 
-		if (!restrained())
-			var/diag = get_dir(src, pulling)
-			if ((diag - 1) & diag)
-			else
-				diag = null
-			if ((get_dist(src, pulling) > 1 || diag))
-				if (isliving(pulling))
-					var/mob/living/M = pulling
-					var/pull_ok = 1
-					if (M.grabbed_by.len)
-						if (prob(75))
-							var/obj/item/weapon/grab/G = pick(M.grabbed_by)
-							visible_message("<span class='danger'>[src] has pulled [G.affecting] from [G.assailant]'s grip.</span>")
-							qdel(G)
-						else
-							pull_ok = 0
-						if (M.grabbed_by.len)
-							pull_ok = 0
-					if (pull_ok)
-						var/atom/movable/t = M.pulling
-						M.stop_pulling()
-
-						//this is the gay blood on floor shit -- Added back -- Skie
-						if(M.lying && !M.buckled && (prob(M.getBruteLoss() / 2)))
-							makeTrail(T, M)
-						pulling.Move(T, get_dir(pulling, T))
-						if(M)
-							M.start_pulling(t)
-				else
-					if (pulling)
-						pulling.Move(T, get_dir(pulling, T))
-	else
-		stop_pulling()
-		. = ..()
+		var/pull_dir = get_dir(src, pulling)
+		if(get_dist(src, pulling) > 1 || ((pull_dir - 1) & pull_dir)) //puller and pullee more than one tile away or in diagonal position
+			if(isliving(pulling))
+				var/mob/living/M = pulling
+				if(M.lying && !M.buckled && (prob(M.getBruteLoss() / 2)))
+					makeTrail(T, M)
+			pulling.Move(T, get_dir(pulling, T)) //the pullee tries to reach our previous position
+			if(pulling && get_dist(src, pulling) > 1) //the pullee couldn't keep up
+				stop_pulling()
+
+	if(pulledby && moving_diagonally != FIRST_DIAG_STEP && get_dist(src, pulledby) > 1)//separated from our puller and not in the middle of a diagonal move.
```

Comment:
The check for diagonal move is needed, otherwise if you start pulling something that's in a diagonal position to you and then move away, the pullee's first step of its diagonal move may not be the one that brings them closer to the puller so it will still be two tiles apart from the puller and the pull will be broken.